src/services/amazonDSP.js

```diff
@@ -0,0 +1,89 @@
+const {
+    validateCredentials,
+    getConfig,
+    createDSPCampaign,
+    getDSPCampaigns,
+} = require('../utils/allocations');
+
+const getAmazonDSPCampaigns = async ({ access, profileId, advertiserId }) => {
+    try {
+        validateCredentials(access);
+
+        const config = getConfig({
+            type: 'Sponsored Ads',
+            access,
+            profileId,
+            method: 'get',
+        });
+
+        // Get campaigns and return responses
+        const campaigns = await getDSPCampaigns({ config, advertiserId });
+
+        return {
+            message: 'Amazon DSP campaign retrieval complete',
+            data: campaigns?.data?.response || [],
+        };
+    } catch (error) {
+        return {
+            message: 'Amazon DSP campaign retrieval failed',
+            error: {
+                code: 500,
+                message: error.message,
+            },
+            data: [],
+        };
+    }
+};
+
+const createAmazonDSPCampaigns = async ({
+    campaigns,
+    profileId,
+    access,
+    advertiserId,
+}) => {
+    try {
+        // Validate credentials
+        validateCredentials(access);
+
+        // validate advertiserId
+        if (!advertiserId) {
+            throw new Error('Advertiser ID is required');
+        }
+
+        const config = getConfig({ type: 'Sponsored Ads', access, profileId });
+
+        // Create campaigns and handle responses
+        const campaignsCreated = [];
+
+        for (const campaign of campaigns) {
+            campaignsCreated.push(
```

Comment:
@PedroLabrador base on what we talk today the try catch block shouldn't be inside the for loop to try to create all the campaigns and and save the errors for the campaigns that wasn't created 

and at the end of the loop return a promise